applications/conceptual-model-editor/src/app/diagram/dialog/select-classes-dialog.tsx

```diff
@@ -0,0 +1,117 @@
+import { t } from "../application/";
+import { isSemanticModelClassUsage } from "@dataspecer/core-v2/semantic-model/usage/concepts";
+import { type EntityDetailSupportedType, EntityProxy } from "../util/detail-utils";
+import { tailwindColorToHex } from "~/app/utils/color-utils";
+
+const getDefaultColor = () => {
+    return "#069420";
+};
+
+export const useSelectClassesDialog = () => {
+    const { isOpen, open, close, BaseDialog } = useBaseDialog();
+    const [onSelectConfirmCallback, setOnSelectConfirmCallback] = useState<((newEntityID: string[]) => void) | undefined>(undefined);
+
+    const localOpen = (onSelectConfirmCallback?: (newEntityID: string[]) => void) => {
+        setOnSelectConfirmCallback(onSelectConfirmCallback);
+        open();
+    };
+
+    // TODO: Maybe could be optimized later, when I get to know React better
+    function createChangedRecord<T>(record: Record<string, T>, id: string, newValue: T): Record<string, T> {
+        const newRecord = {...record};
+        newRecord[id] = newValue;
+        return newRecord;
+    }
+
+
+    const SelectClassesDialog = () => {   
+        const [connectionsMap, setConnectionsMap] = useState<Record<string, boolean>>({});
+
+        const { models, aggregatorView } = useModelGraphContext();
+        const activeVisualModel = aggregatorView.getActiveVisualModel();     
+
+        const { language: preferredLanguage } = useConfigurationContext();
+
+        const getPreferredName = useCallback((entity: EntityDetailSupportedType) => {
+            const { name } = EntityProxy(entity, preferredLanguage);
+            return name;
+        }, [preferredLanguage]);
+            
+        
+        // TODO: Probably could use better dependecies, but if it works it works
+        const getSelectedClasses = useCallback(() => {
+            const selectedClasses = Object.entries(connectionsMap).map(([k, v]) => {                
+                if(v) {
+                    return k;
+                }
+            });
+            if(selectedClasses === undefined) {
+                return [];
+            }
+            else {
+                return selectedClasses.filter(e => e !== undefined);
+            }
+        }, [connectionsMap]);
+
+        const onAcceptCallback = useCallback(() => {
+            if(onSelectConfirmCallback === undefined) {                
+                close();                
+            }
+            else {
+                onSelectConfirmCallback(getSelectedClasses());
+                close();
+            }            
+        }, [getSelectedClasses]);
+           
+
+        // TODO: Probably will have to show profiles later differently (like in catalog - using TreeLikeOffset)
+        return (
+            <BaseDialog heading={t("select-classes-dialog.select-classes")}>
+                <div>
+                    {/* We could also modify and then reuse <EntityCatalog />, it would be kind of nice, but it would take a lot of time */}
```

Comment:
This is not even a TODO ..  use comments to explain code, intentions and limitations. Do not use comments to plan future work.